### DISCUSSION TITLE
select选择器能否显示带icon的内容？

### DISCUSSION THREAD

**Discussion by FatNerdPeng** [element-plus/element-plus#13492] [Q&A] [answered]
家人们，我希望能达到这个效果：
![image](https://github.com/element-plus/element-plus/assets/33616855/dc1bef89-6aa7-41ee-b5ea-306a7a08fb28)
但是我现在只能达到这个效果：
![image](https://github.com/element-plus/element-plus/assets/33616855/03b5b3be-ca49-4b9d-bc73-5ec14071d3f5)
这是我的代码：
![image](https://github.com/element-plus/element-plus/assets/33616855/ab0d912c-43de-4104-96c8-b767eae953e8)
![image](https://github.com/element-plus/element-plus/assets/33616855/d635e5ff-2945-4c71-858d-f97a9991f8c3)
能实现吗？能的话我应该如何改进？谢谢

**Comment by FatNerdPeng** [comment]
上面那个图截错了，我现在的效果是：
![image](https://github.com/element-plus/element-plus/assets/33616855/2e3b6b21-b558-4aaa-9ca5-21206aa7afa1)
就是我选中了一个项后，它只显示文字

**Comment by ryuhangyeong** [comment]
~~this is impossible as far as I know.~~

**Comment by markhermano** [comment, marked as answer]
@FatNerdPeng 
Try this:
```html
<template>
  <el-select v-model="value" placeholder="Select" size="large">
    <template #prefix>
      {{ value.flag }}
    </template>
    <el-option
      v-for="item in options"
      :key="item.value"
      :label="item.label"
      :value="item"
    >
      {{ item.flag }}&nbsp;&nbsp;{{ item.label }}
    </el-option>
  </el-select>
</template>

<script lang="ts" setup>
import { ref } from 'vue'

const options = [
  {
    value: 'EN',
    label: 'US',
    flag: '🇺🇸'
  },
  {
    value: 'ES',
    label: 'Español',
    flag: '🇪🇸'
  },
  {
    value: 'ZH',
    label: '简体中文',
    flag: '🇨🇳'
  },
]
const value = ref(options[2])
</script>
```
[live example here](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXNlbGVjdCB2LW1vZGVsPVwidmFsdWVcIiBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiIHNpemU9XCJsYXJnZVwiPlxuICAgIDx0ZW1wbGF0ZSAjcHJlZml4PlxuICAgICAge3sgdmFsdWUuZmxhZyB9fVxuICAgIDwvdGVtcGxhdGU+XG4gICAgPGVsLW9wdGlvblxuICAgICAgdi1mb3I9XCJpdGVtIGluIG9wdGlvbnNcIlxuICAgICAgOmtleT1cIml0ZW0udmFsdWVcIlxuICAgICAgOmxhYmVsPVwiaXRlbS5sYWJlbFwiXG4gICAgICA6dmFsdWU9XCJpdGVtXCJcbiAgICA+XG4gICAgICB7eyBpdGVtLmZsYWcgfX0mbmJzcDsmbmJzcDt7eyBpdGVtLmxhYmVsIH19XG4gICAgPC9lbC1vcHRpb24+XG4gIDwvZWwtc2VsZWN0PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIHZhbHVlOiAnRU4nLFxuICAgIGxhYmVsOiAnVVMnLFxuICAgIGZsYWc6ICfwn4e68J+HuCdcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAnRVMnLFxuICAgIGxhYmVsOiAnRXNwYcOxb2wnLFxuICAgIGZsYWc6ICfwn4eq8J+HuCdcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAnWkgnLFxuICAgIGxhYmVsOiAn566A5L2T5Lit5paHJyxcbiAgICBmbGFnOiAn8J+HqPCfh7MnXG4gIH0sXG5dXG5jb25zdCB2YWx1ZSA9IHJlZihvcHRpb25zWzJdKVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

**Reply by FatNerdPeng** [reply]
Thanks, your code works!

**Reply by markhermano** [reply]
I'm glad it works for you. I hope this answers your question.

**Reply by ryuhangyeong** [reply]
@markhermano Good!

**Reply by faga295** [reply]
Good!

**Reply by ryuhangyeong** [reply]
@faga295 sorry i think it was tagged wrong 😅
